### DISCUSSION TITLE
Get total number of records in the collection

### DISCUSSION THREAD

**Discussion by speryt** [pocketbase/pocketbase#4666] [Q&A] [answered]
Is there a way other than `getFullList()` to obtain **number** of records in the collection?
For example `getList()` has page as an argument, but without full list it seems impossible to generate proper pagination interface. Not mentioning case where collection has thousands of records and we might not want to send all of them to the user.

Or am I just missing something and this would not be a concern?

**Comment by ganigeorgiev** [comment, marked as answer]
`getList()` returns the following fields:

```js
page         // current page
perPage      // current page
totalItems   // total number of items
totalPages   // total number of pages
items        // the items in the current page
```

So you should be able to use `totalItems` and `totalPages` to create your pagination interface.

Note that if you are sending with the request `skipTotal=1` query param then the above 2 fields will be always `-1` (this is more performant as it avoid the extra COUNT query and it is suitable for "infinity"/cursor type of scroll/pagination).

**Reply by speryt** [reply]
Thank you very much for clarification. I must have completely missed that part from documentation. My bad.
